cdb2api/cdb2api.c

```diff
@@ -3267,34 +3304,44 @@ static int cdb2_run_statement_typed_int(cdb2_hndl_tp *hndl, const char *sql,
     }
 
     if (type == RESPONSE_HEADER__SQL_RESPONSE_SSL) {
+#if WITH_SSL
         hndl->s_sslmode = PEER_SSL_REQUIRE;
-        try_ssl(hndl, hndl->sb, hndl->connected_host);
+        /* server wants us to use ssl so turn ssl on in same connection */
+        int resp = try_ssl(hndl, hndl->sb, hndl->connected_host);
```

Comment:
We should always retry the query in case the node crashes during the connection attempt.